notebooks/Airline Delays - Final Writeup

```diff
@@ -234,11 +288,66 @@ ORDER BY origin_aa1_rain_depth"""))
+
+# COMMAND ----------
+
+#Review summary stats for weather delays 
+display(train_log.filter(col('weather_15_delay_indicator')==1).select('origin_WND_speed_rate',"origin_TMP_air_temperature",'origin_CIG_ceiling_height',"origin_DEW_dew_point_temp",'origin_VIS_distance','origin_SLP_sea_level_pressure','origin_aj1_snow_depth','origin_aa1_rain_depth').summary())
+
+#Review summary stats for other non-weather delayed flights 
+display(train_log.filter(col('weather_15_delay_indicator')==0).select('origin_WND_speed_rate',"origin_TMP_air_temperature",'origin_CIG_ceiling_height',"origin_DEW_dew_point_temp",'origin_VIS_distance','origin_SLP_sea_level_pressure','origin_aj1_snow_depth','origin_aa1_rain_depth').summary())
+
+# COMMAND ----------
+
+summer_months=[5,6,7,8,9,10]
+winter_months=[1,2,3,4,11,12]
+
+#Review histogram of air temperatures associated with weather delays (1) and flights with no weather delay (0) during the summer
+train_log.filter(~col('month').isin(summer_months)).select('origin_TMP_air_temperature',"weather_15_delay_indicator").display()
+
+# COMMAND ----------
+
 # MAGIC %md
 # MAGIC ## Algorithm Exploration
 
 # COMMAND ----------
 
+# MAGIC %md
+# MAGIC #### Base Model - Logistic Regression
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC Our base model is a logistic regression classification model with a subset of the airline and weather features. We included features to capture seasonality, recent weather patterns at the origin airport, time of day, as well as two engineered features that captured whether a previous flight was delayed and whether the day of the flight was a holiday. During our EDA we identified that dew point and air temp were strongly correlated so we only included air temperature to capture the most recent temperature. When we ran our initial logistic regression model, we observed that the model always predicted that there was no delay due to the imbalance in the dataset with under 20% of flights being delayed. To account for the imbalance of the outcome variable we explored two options for the base model including a weight column that utilizes a balancing ratio and adjusting the threshold for the model to a value closer to the mix of delays to no delays. The threshold was slightly more effective at improving recall for our model than the weighted column so we utilitized the threshold in our base model which is shown below. As we increased the threshold from .18 to .5 we noticed a large drop in recall with an increase in accuracy, we found that a threshold below .2 had good balance of accuracy and recall. The most noticeable improvement in the model accuracy and recall came when we added the previous flight delay feature, which signaled to us that this feature maybe very important to include in future models. Our base model had an accuracy and recall of **XX and XX (Pending Model Updates**). In our next models we explore using a broader set of the features available as well as explore oversampling.
```

Comment:
"In our next models we explore using a broader set of the features available as well as explore oversampling." - remove second "explore" or maybe replace with something like "additional methods for handling"